### ISSUE TITLE
"this" in arrow functions not working properly

### ISSUE THREAD

**Issue by genexliu** [tststs/atom-ternjs#212]
sample code:
```javascript
// tested on atom 1.5.4 with atom-ternjs 0.13.2 installed
class a {
  foo() {}
  bar() {
    this.foo();   // jumps to foo
    [].forEach(() => {
      this.foo(); // does not jump to foo
    });
  }
}
```

I pasted the same code to the live demo http://ternjs.net/doc/demo/#ES6 but both of the `this.foo()` calls work on the live demo, so this issue might be atom specific.

**Comment by genexliu** [comment]
Linking the same issue that I raised on https://github.com/ternjs/tern : https://github.com/ternjs/tern/issues/730

It looks like that updating to terns 0.18 should fix this.

**Comment by tststs** [comment]
Should be fixed with the release of 0.14.0.
